opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -218,11 +258,24 @@ public SpanBuilder withStartTimestamp(long microseconds) {
 
         @Override
         public MockSpan start() {
+            return startManual();
+        }
+
+        @Override
+        public ActiveSpan startActive() {
+            MockSpan span = this.startManual();
+            return spanSource.makeActive(span);
+        }
+
+        @Override
+        public MockSpan startManual() {
             if (this.startMicros == 0) {
                 this.startMicros = MockSpan.nowMicros();
             }
-            return new MockSpan(MockTracer.this, this.operationName, this.startMicros, initialTags, this.firstParent);
+            if (firstParent == null && !ignoringActiveSpan) {
+                firstParent = (MockSpan.MockContext)activeSpanContext();
```

Comment:
implicit parent should be derived at `tracer.buildSpan()` time https://github.com/opentracing/opentracing-java/pull/115/files#diff-9f484babc6e66bd62198ce009d9247dcR26